### PR TITLE
Enhancement：Support more type value as parameter

### PR DESCRIPTION
**Pull request by CorvusYe**

### CODE REVIEW

**Comment by wey-gu** [comment]
@Nicole00 @CorvusYe could we move this PR on?

**Comment by CorvusYe** [comment]
@wey-gu 
There are some conflicts. 
I will resolve tomorrow, if it is still helpful in this version.

File: client/src/main/java/com/vesoft/nebula/client/graph/net/Session.java [hunk 1]
```diff
@@ -402,55 +396,112 @@ private NMap map2Nmap(Map<String, Object> map) throws UnsupportedOperationExcept
...
+    }
+
+    /**
+     * some value setter for java type (basic or nebula special type) that need convert to NValue
+     */
+    public static Map<Class<?>, Setter> LEAF_TYPE_AND_SETTER = new HashMap<Class<?>, Setter>() {{
+        put(Value.class, (Setter<Value>) (param) -> param );
+        put(boolean.class, (Setter<Boolean>) Value::bVal);
+        put(Boolean.class, (Setter<Boolean>) Value::bVal);
+        put(int.class, (Setter<Integer>) Value::iVal);
```
Comment (Nicole00) [inline client/src/main/java/com/vesoft/nebula/client/graph/net/Session.java:428]: is it necessary to put the base data type class? such as int.class.

File: client/src/main/java/com/vesoft/nebula/util/ReflectUtil.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
+// Copyright 2022-present Weicheng Ye. All rights reserved.
```
Comment (Nicole00) [inline client/src/main/java/com/vesoft/nebula/util/ReflectUtil.java:1]: please update the copyright

File: client/src/main/java/com/vesoft/nebula/client/graph/net/Session.java (hunk 1 above)
Comment (CorvusYe) [reply client/src/main/java/com/vesoft/nebula/client/graph/net/Session.java:428]: > is it necessary to put the base data type class? such as int.class.

Yes, the hashcode is different between `Integer.class` and `int.class`

Comment (CorvusYe) [reply client/src/main/java/com/vesoft/nebula/client/graph/net/Session.java:428]: > > is it necessary to put the base data type class? such as int.class.
> 
> Yes, the hashcode is different between `Integer.class` and `int.class`

Oh... It is unnecessary... when int auto sealing, intVal.getClass() return Integer. base data type will never be visited
You are right. @Nicole00

**Comment by CorvusYe** [comment]
@Nicole00 
I have another question: 
  Does the current version require java8? Can I use lamda syntax as before?

**Comment by Nicole00** [comment]
> @Nicole00 I have another question: Does the current version require java8? Can I use lamda syntax as before?

Yeah, we require java 8.

**Pull request closed without merging**